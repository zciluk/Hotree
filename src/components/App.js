import React from "react";
import "../style/main.scss";
import FormContainer from "./FormContainer.js";
import Header from "./Header";

const App = props => (
      <div className="container">
        <Header />
        <FormContainer />
      </div>
    );


export default App;
