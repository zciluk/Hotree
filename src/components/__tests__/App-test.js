import React from 'react';
import App from '../App.js';
import renderer from 'react-test-renderer';
Date.now = jest.fn(() => 1709915676000); //some date in 2024

// basic snapshot test of initial layout render - checks on every run if layout, UI and components are in place
test('should render initial layout', () => {
    const component = renderer.create(
        <App/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
   
});