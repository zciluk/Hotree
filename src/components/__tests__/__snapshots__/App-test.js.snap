// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render initial layout 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <div
      className="header__small-line"
    />
    <div
      className="header__big-box"
    >
      <span
        className="header__big-box--text"
      >
        New event
      </span>
    </div>
  </header>
  <div>
    <div
      className="form-container false"
    >
      <div
        className="box-container"
      >
        <div
          className="box-container__title"
        >
          About
        </div>
        <hr
          className="box-container__vertical"
        />
        <div
          className="field-container"
        >
          <label
            className="field-container__label false"
          >
            Title
            <label
              className="field-container__label--required"
            >
              *
            </label>
          </label>
          <div
            className="field-container__field"
          >
            <input
              className="form-input false"
              id="title"
              onChange={[Function]}
              placeholder="Make it short and clear"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="field-container"
        >
          <label
            className="field-container__label false"
          >
            Description
            <label
              className="field-container__label--required"
            >
              *
            </label>
          </label>
          <div
            className="field-container__field"
          >
            <div
              className="form-flex"
            >
              <textarea
                className="form-textarea false"
                id="description"
                onChange={[Function]}
                placeholder="Write about your event, be creative"
                type="text"
                value=""
              />
              <label
                className="description-item"
              >
                Max length: 
                140
                 characters
                 
              </label>
              <label
                className="description-item description-item__left false"
              >
                0
                /
                140
              </label>
            </div>
          </div>
        </div>
        <div
          className="field-container"
        >
          <label
            className="field-container__label undefined"
          >
            Category
          </label>
          <div
            className="field-container__field"
          >
            <div
              className="form-flex"
            >
              <select
                className="form-select false"
                id="select"
                onChange={[Function]}
                value=""
              >
                <option
                  disabled={true}
                  value=""
                >
                  Select category
                </option>
                <option
                  value={0}
                >
                  Cycling
                </option>
                <option
                  value={1}
                >
                  Hiking
                </option>
                <option
                  value={2}
                >
                  Cooking
                </option>
                <option
                  value={3}
                >
                  Rock climbing
                </option>
                <option
                  value={4}
                >
                  Yoga
                </option>
                <option
                  value={5}
                >
                  Fencing
                </option>
                <option
                  value={6}
                >
                  Swimming
                </option>
                <option
                  value={7}
                >
                  Badminton
                </option>
                <option
                  value={8}
                >
                  Running
                </option>
                <option
                  value={9}
                >
                  Dance
                </option>
              </select>
              <label
                className="description-item"
              >
                Describes topic and people who should be interested in this event
              </label>
            </div>
          </div>
        </div>
        <div
          className="field-container"
        >
          <label
            className="field-container__label false"
          >
            Payment
          </label>
          <div
            className="field-container__field"
          >
            <div
              className="form-payment"
            >
              <div
                className="form-payment__radio-group"
              >
                <input
                  checked={true}
                  className="form-payment__radio-input"
                  id="free"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="form-payment__radio-label"
                  htmlFor="free"
                >
                  <span
                    className="form-payment__radio-button"
                  />
                  <span
                    className="form-payment__radio-labelname"
                  >
                    Free event
                  </span>
                </label>
              </div>
              <div
                className="form-payment__radio-group"
              >
                <input
                  checked={false}
                  className="form-payment__radio-input"
                  id="paid"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="form-payment__radio-label"
                  htmlFor="paid"
                >
                  <span
                    className="form-payment__radio-button"
                  />
                  <span
                    className="form-payment__radio-labelname"
                  >
                    Paid event
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          className="field-container"
        >
          <label
            className="field-container__label undefined"
          >
            Reward
          </label>
          <div
            className="field-container__field"
          >
            <div
              className="form-smallinput"
            >
              <input
                className="form-smallinput__input "
                id="reward"
                onChange={[Function]}
                placeholder="Number"
                type="number"
                value=""
              />
              <span
                className="form-smallinput__label"
              >
                rewards points for attendance
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="box-container"
      >
        <div
          className="box-container__title"
        >
          Coordinator
        </div>
        <hr
          className="box-container__vertical"
        />
        <div
          className="field-container"
        >
          <label
            className="field-container__label undefined"
          >
            Responsible
            <label
              className="field-container__label--required"
            >
              *
            </label>
          </label>
          <div
            className="field-container__field"
          >
            <div
              className="form-flex"
            >
              <select
                className="form-select form-select__clicked"
                id="coordinator"
                onChange={[Function]}
                value={3}
              >
                <option
                  disabled={true}
                  value=""
                >
                  Me
                </option>
                <option
                  value={3}
                >
                  Me - 
                  Walter
                   
                  Nelson
                </option>
                <option
                  disabled={true}
                  value=""
                >
                  Others
                </option>
                <option
                  value={0}
                >
                  Daniel
                   
                  Mitchell
                </option>
                <option
                  value={1}
                >
                  Albert
                   
                  Alexander
                </option>
                <option
                  value={2}
                >
                  Philip
                   
                  Hughes
                </option>
                <option
                  value={4}
                >
                  Ashley
                   
                  Hernandez
                </option>
                <option
                  value={5}
                >
                  Donna
                   
                  Washington
                </option>
                <option
                  value={6}
                >
                  Andrew
                   
                  White
                </option>
                <option
                  value={7}
                >
                  Sharon
                   
                  Allen
                </option>
                <option
                  value={8}
                >
                  Russell
                   
                  Parker
                </option>
                <option
                  value={9}
                >
                  Janet
                   
                  Stewart
                </option>
              </select>
              <label
                className="description-item"
              >
                
              </label>
            </div>
          </div>
        </div>
        <div
          className="field-container"
        >
          <label
            className="field-container__label false"
          >
            E-mail
          </label>
          <div
            className="field-container__field"
          >
            <input
              className="form-input false"
              id="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="box-container"
      >
        <div
          className="box-container__title"
        >
          When
        </div>
        <hr
          className="box-container__vertical"
        />
        <div
          className="field-container"
        >
          <label
            className="field-container__label false"
          >
            Starts on
            <label
              className="field-container__label--required"
            >
              *
            </label>
          </label>
          <div
            className="field-container__field"
          >
            <div
              className="form-date"
            >
              <input
                className="form-date__datepicker false"
                id="date"
                min="2024-03-08"
                onChange={[Function]}
                placeholder="dd/mm/yyyy"
                type="date"
                value=""
              />
              <span
                className="form-date__label"
              >
                 at 
              </span>
              <input
                className="form-date__timepicker false"
                id="time"
                max="12:00"
                min="0:00"
                onChange={[Function]}
                placeholder="--:--"
                type="time"
                value=""
              />
              <div
                className="form-payment__radio-group"
              >
                <input
                  checked={true}
                  className="form-payment__radio-input"
                  id="am"
                  name="AM"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="form-payment__radio-label"
                  htmlFor="am"
                >
                  <span
                    className="form-payment__radio-button"
                  />
                  <span
                    className="form-payment__radio-labelname"
                  >
                    AM
                  </span>
                </label>
              </div>
              <div
                className="form-payment__radio-group"
              >
                <input
                  checked={false}
                  className="form-payment__radio-input"
                  id="pm"
                  name="PM"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="form-payment__radio-label"
                  htmlFor="pm"
                >
                  <span
                    className="form-payment__radio-button"
                  />
                  <span
                    className="form-payment__radio-labelname"
                  >
                    PM
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          className="field-container"
        >
          <label
            className="field-container__label undefined"
          >
            Duration
          </label>
          <div
            className="field-container__field"
          >
            <div
              className="form-smallinput"
            >
              <input
                className="form-smallinput__input "
                id="duration"
                onChange={[Function]}
                placeholder="Number"
                type="number"
                value=""
              />
              <span
                className="form-smallinput__label"
              >
                hours
              </span>
            </div>
          </div>
        </div>
      </div>
      <button
        className="form-button"
        id="submit"
        onClick={[Function]}
      >
        Publish event
      </button>
    </div>
  </div>
</div>
`;
